uplink/src/collector/tcpjson.rs

```diff
@@ -70,29 +78,48 @@ impl Bridge {
                     }
                     action = self.actions_rx.recv_async() => {
                         let action = action?;
-                        error!("Bridge down!! Action ID = {}", action.action_id);
-                        let status = ActionResponse::failure(&action.action_id, "Bridge down");
-                        if let Err(e) = self.action_status.fill(status).await {
-                            error!("Failed to send busy status. Error = {:?}", e);
+                        let action_id = action.action_id.clone();
+                        if self.actions_tx.send(action).is_err() {
+                            error!("Bridge down!! Action ID = {}", action_id);
+                            let status = ActionResponse::failure(&action_id, "Bridge down");
+                            if let Err(e) = self.action_status.fill(status).await {
+                                error!("Failed to send busy status. Error = {:?}", e);
+                            }
                         }
                     }
                 }
-            };
+            }
+        }
+    }
 
-            info!("Accepted new connection from {:?}", addr);
-            let framed = Framed::new(stream, LinesCodec::new());
-            if let Err(e) = self.collect(framed).await {
+    async fn spawn_collector(&self, stream: TcpStream) {
+        let framed = Framed::new(stream, LinesCodec::new());
+        let mut tcp_json = TcpJson {
+            action_status: self.action_status.clone(),
+            streams: Streams::new(self.config.clone(), self.data_tx.clone()),
+            actions_rx: self.actions_tx.subscribe(),
+        };
+        tokio::task::spawn(async move {
+            if let Err(e) = tcp_json.collect(framed).await {
                 error!("Bridge failed. Error = {:?}", e);
             }
-        }
+        });
     }
+}
 
+struct TcpJson {
+    streams: Streams,
+    action_status: Stream<ActionResponse>,
+    actions_rx: BRx<Action>,
+}
+
+impl TcpJson {
+    /// NOTE: We only expect one action to be processed over uplink's bridge at a time,
```

Comment:
This is a constraint that clients have to follow too. Add this to apps.md as well.